pkg/delegate_test.go

```diff
@@ -5,56 +5,132 @@ import (
 	})
-	assert.NoError(t, err)
 
-	err = d.Start(context.Background())
-	assert.Equal(t, err.Error(), "unimplemented")
+	t.Run("creates the delegate with the provided config", func(t *testing.T) {
+		logger := ktypes.NewMockLogger(t)
+		logger.On("Debug", mock.Anything, mock.Anything).Maybe()
+
+		_, err := NewDelegate(DelegateConfig{
+			Logger: logger,
+			LocalConfig: types.LocalConfig{
+				BlockchainTimeout:                  1 * time.Second,
+				ContractConfigTrackerPollInterval:  15 * time.Second,
+				ContractTransmitterTransmitTimeout: 1 * time.Second,
+				DatabaseTimeout:                    100 * time.Millisecond,
+				ContractConfigConfirmations:        1,
+			},
+			CacheExpiration:       1 * time.Second,
+			CacheEvictionInterval: 1 * time.Second,
+			MaxServiceWorkers:     1,
+			ServiceQueueLength:    1,
+		})
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("running the delegate runs the wrapped keeper", func(t *testing.T) {
+		var started bool
+		delegate := &Delegate{
+			keeper: &mockKeeper{
+				StartFn: func() error {
+					started = true
+					return nil
+				},
+			},
+		}
+		err := delegate.Start(context.Background())
+		assert.NoError(t, err)
+		assert.True(t, started)
+	})
+
+	t.Run("starting the delegate errors when the keeper errors on start", func(t *testing.T) {
+		var started bool
+		delegate := &Delegate{
+			keeper: &mockKeeper{
+				StartFn: func() error {
+					started = true
+					return errors.New("failed to start")
+				},
+			},
+		}
+		err := delegate.Start(context.Background())
+		assert.Error(t, err)
+		assert.True(t, started)
+	})
 }
 
 func TestClose(t *testing.T) {
-	var err error
-	var mockLogger = ktypes.NewMockLogger(t)
-	mockLogger.On("Debug", mock.Anything, mock.Anything).Return()
-
-	d, err := NewDelegate(DelegateConfig{
-		Logger: mockLogger,
-		LocalConfig: types.LocalConfig{
-			BlockchainTimeout:                  1 * time.Second,
-			ContractConfigTrackerPollInterval:  15 * time.Second,
-			ContractTransmitterTransmitTimeout: 1 * time.Second,
-			DatabaseTimeout:                    100 * time.Millisecond,
-			ContractConfigConfirmations:        1,
-		},
+	t.Run("a not yet started oracle fails to close", func(t *testing.T) {
+		var mockLogger = ktypes.NewMockLogger(t)
+		mockLogger.On("Debug", mock.Anything, mock.Anything).Return()
+
+		d, err := NewDelegate(DelegateConfig{
+			Logger: mockLogger,
+			LocalConfig: types.LocalConfig{
+				BlockchainTimeout:                  1 * time.Second,
+				ContractConfigTrackerPollInterval:  15 * time.Second,
+				ContractTransmitterTransmitTimeout: 1 * time.Second,
+				DatabaseTimeout:                    100 * time.Millisecond,
+				ContractConfigConfirmations:        1,
+			},
+		})
+		assert.NoError(t, err)
+		if err != nil {
+			t.FailNow()
+		}
```

Comment:
is failNow some custom shutdown?

Reply:
So I guess that was added as a part of the original unit tests, but it looks to be what the assert library calls under the hood to fail tests? I'll update this to use your suggestion though since I think that's cleaner! 🙌